index.bs

```diff
@@ -499,6 +548,97 @@ Modify the <a spec=HTML>downloads a hyperlink</a> algorithm to ensure that downl
+
+<h4 id="patch-geolocation">Geolocation API</h4>
+
+<div algorithm="Geolocation getCurrentPosition patch">
+  Modify the {{Geolocation/getCurrentPosition()}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=] and return.
+</div>
+
+<div algorithm="Geolocation watchPosition patch">
+  Modify the {{Geolocation/watchPosition()}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then:
+
+    1. Let |watchProcessId| be a new unique <a spec=GEOLOCATION>watch process</a> ID, and note it as a <dfn>post-prerendering activation geolocation watch process ID</dfn>.
+
+    1. Append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=], with the modification that the <a spec=GEOLOCATION>watch process</a> generated must use |watchProcessId| as its ID, and return |watchProcessId|.
+</div>
+
+<div algorithm="Geolocation clearWatch patch">
+  Modify the {{Geolocation/clearWatch(watchId)}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then:
+
+    1. If <var ignore>watchId</var> is a [=post-prerendering activation geolocation watch process ID=], then remove its corresponding steps from [=this=]'s [=platform object/post-prerendering activation steps list=].
+
+    1. Return.
+</div>
+
+<h4 id="patch-serial">Web Serial API</h4>
+
+<div algorithm="Serial requestPort patch">
+  Modify the {{Serial/requestPort()}} method steps by inserting the following steps after the initial creation of |promise|:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=] and return |promise|.
+</div>
+
+<div algorithm="Serial getPorts patch">
+  Modify the {{Serial/getPorts()}} method steps by inserting the following steps after the initial creation of |promise|:
+
+  <!-- TODO link owner document once https://github.com/whatwg/html/pull/6379 lands -->
+  1. Let |document| be [=this=]'s [=relevant global object=]'s [=associated Document=], if [=this=]'s [=relevant global object=] is a {{Window}}, or [=this=]'s [=relevant global object=]'s owner document, if [=this=]'s [=relevant global object=] is a {{DedicatedWorkerGlobalScope}}.
```

Comment:
Is there any reason not to make the relevant the associated Document be the owner document for a window global? It feels like this complication need not live at the use of this term.

Reply:
Hmm. I guess I think they're sufficiently different things that they shouldn't use the same term. A Window <-> Document relationship is rather special, but the Window -> DedicatedWorkerGlobalScope ownership feels like a different sort of thing. Also, since it would only work for Window and (some) DedicatedWorkerGlobalScopes, not other globals, the generalization doesn't buy us that much.